app/grandchallenge/profiles/migrations/0019_migrate_mfa.py

```diff
@@ -0,0 +1,50 @@
+import base64
+
+from allauth.mfa.adapter import get_adapter
+from allauth.mfa.models import Authenticator
+from django.db import migrations
+from django_otp.plugins.otp_static.models import StaticDevice
+from django_otp.plugins.otp_totp.models import TOTPDevice
+
+
+def migrate_mfa(apps, schema_editor):
+    adapter = get_adapter()
+    authenticators = []
+    for totp in TOTPDevice.objects.filter(confirmed=True).iterator():

```

Comment:
I trust this migration has been tested? e.g. creating a TOTP device with the old code, then migrating, then logging in with the new. Same goes for recovery codes.

Reply:
Yes, all tested and works.